pyfr/solvers/baseadvec/system.py

```diff
@@ -105,12 +105,14 @@ def deps(dk, *names): return self._kdeps(k, dk, *names)
         g1 = self.backend.graph()
         g1.add_mpi_reqs(m['ent_fpts_recv'])
 
+        g1.add_all(k['eles/entropy_filter'])
```

Comment:
Is this always needed?

Reply:
Sort of. For open upts, then yes to ensure that your solution at the fpts is positivity-preserving to avoid an instant NaN. Same with closed upts if you have L2 projection (which doesn't necessarily ensure that the solution is positive after projection). For closed upts without L2 projection, it technically doesn't require it, but in that case all points are positivity-preserving (assuming that your ICs are positive) and then it just skips the filtering step.